docs/examples/Reinforcement_Learning_Sensor_Manager.py

```diff
@@ -0,0 +1,1002 @@
+predictor = KalmanPredictor(transition_model)
+
+from stonesoup.updater.kalman import ExtendedKalmanUpdater
+updater = ExtendedKalmanUpdater(measurement_model=None)
+# measurement model is added to detections by the sensor
+
+# %%
+# Run the Kalman filters
+# ----------------------
+# First create `ntruths` priors which estimate the targets’ initial states, one for each target. In this example
+# each prior is offset by 0.5 in the y direction meaning the position of the track is initially not very accurate. The
+# velocity is also systematically offset by +0.5 in both the x and y directions.
+# 
+# 
+
+from stonesoup.types.state import GaussianState
+
+priors = []
+xdirection = 1.2
+ydirection = 1.2
+for j in range(0, ntruths):
+    priors.append(GaussianState([[0], [xdirection], [yps[j]+0.1], [ydirection]],
+                                np.diag([0.5, 0.5, 0.5, 0.5]+np.random.normal(0,5e-4,4)),
+                                timestamp=start_time))
+    xdirection *= -1
+    if j % 2 == 0:
+        ydirection *= -1
+
+
+# %%
+# Initialise the tracks by creating an empty list and appending the priors generated. This needs to be done separately
+# for both sensor manager methods as they will generate different sets of tracks.
+# 
+
+from stonesoup.types.track import Track
+
+# Initialise tracks from the RandomSensorManager
+tracksA = []
+for j, prior in enumerate(priors):
+    tracksA.append(Track([prior]))
+
+tracksB = []
+for j, prior in enumerate(priors):
+    tracksB.append(Track([prior]))
+
+# %%
+# Reward function
+# ---------------
+# A reward function is used to quantify the benefit of sensors taking a particular action or set of actions.
+# This can be crafted specifically for an example in order to achieve a particular objective. The function used in
+# this example is quite generic but could be substituted for any callable function which returns a numeric
+# value that the sensor manager can maximise.
+# 
+# The :class:`~.UncertaintyRewardFunction` calculates the uncertainty reduction by computing the difference between the
+# covariance matrix norms of the prediction, and the posterior assuming a predicted measurement corresponding to that
+# prediction.
+
+from stonesoup.hypothesiser.distance import DistanceHypothesiser
+from stonesoup.measures import Mahalanobis
+hypothesiser = DistanceHypothesiser(predictor, updater, measure=Mahalanobis(), missed_distance=5)
+
+from stonesoup.dataassociator.neighbour import GNNWith2DAssignment
+data_associator = GNNWith2DAssignment(hypothesiser)
+
+from stonesoup.sensormanager.reward import UncertaintyRewardFunction
+reward_function = UncertaintyRewardFunction(predictor=predictor, updater=updater)
+
+# %%
+# Reinforcement Learning
+# ----------------------
+# Reinforcement learning is when an intelligent agents make decisions to maximise a cumulative reward. First, the agent
+# must train in an environment to create a policy, which determines which actions to take. During training, the agent
```

Comment:
```suggestion
# must train in an environment in order to create a policy, which later determines the actions it will take. During training, the agent
```